src/java/org/apache/cassandra/cql3/PasswordObfuscator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Obfuscates passwords in a given string
+ */
+public class PasswordObfuscator
+{
+    public static final String OBFUSCATION_TOKEN = "*******";
+    private static final String PASSWORD_TOKEN = "password";
+
+    private static final int PATTERN_FLAGS = Pattern.CASE_INSENSITIVE | Pattern.DOTALL;
+
+    private static final Pattern PASSWORD_PATTERN = Pattern.compile(".*password\\s*=?\\s*'(?<password>.*)'.*", PATTERN_FLAGS);
```

Comment:
Another idea came to my mind with regard to the following use case/statement:
```
CREATE ROLE user1 WITH OPTIONS = { 'custom_option1' : 'password', 'custom_option2' : 99 } AND PASSWORD 'pwd';
```
I guess the current logic won't be working properly (meaning `', '` part will be obfuscated). Could you, please, add the corresponding test to prove / reject the assumption.

Reply:
I think there might be even more use cases and I am wondering whether really the regex is the best option. I need to test more though.

Reply:
Replacing regex with a simpler approach (initial version of the patch) of obfuscating everything after password keyword, similar to what postgres audit does